src/transaction.js

```diff
@@ -93,15 +127,23 @@ Transaction.prototype.isCoinbase = function () {
   return this.ins.length === 1 && Transaction.isCoinbaseHash(this.ins[0].hash)
 }
 
-var EMPTY_SCRIPT = new Buffer(0)
-
 Transaction.prototype.addInput = function (hash, index, sequence, scriptSig) {
+  return this._addInput(hash, index, sequence, scriptSig, undefined)
+}
+
+Transaction.prototype.addWitnessInput = function (hash, index, sequence, witness) {
+  return this._addInput(hash, index, sequence, undefined, witness)
+}
+
+Transaction.prototype._addInput = function (hash, index, sequence, scriptSig, witness) {
   typeforce(types.tuple(
     types.Hash256bit,
     types.UInt32,
     types.maybe(types.UInt32),
+    types.maybe(types.Buffer),
     types.maybe(types.Buffer)
   ), arguments)
+  if (scriptSig && witness) throw TypeError('scriptSig must be blank if witness is provided')
 
```

Comment:
Same problem, scriptSig's aren't ALWAYS blank now. For P2WPKH and P2WSH, the scriptSig's are empty *unless* it's pay-to-script-hash, in which case the witnessProgram is the only element in the scriptSig.